crates/task-impls/src/consensus.rs

```diff
@@ -680,7 +712,6 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                 };
 
                 // promote lock here to add proposal to statemap
-                let mut consensus = RwLockUpgradableReadGuard::upgrade(consensus).await;
```

Comment:
I can't comment on the unchanged line, but I think we could remove the `consensus.high_qc = high_qc` update since it's done in line 501 now?